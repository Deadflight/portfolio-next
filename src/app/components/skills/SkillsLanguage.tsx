import { Icon } from "@/shared/components/Icons/Icons";
import React from "react";

/**
 * Renders a section displaying language proficiency information.
 *
 * This component shows a card with two language entries: Spanish and English.
 * Each entry displays a star rating, language name, proficiency level, and a brief description.
 * The section is accessible with appropriate ARIA labels and headings.
 *
 * @component
 * @example
 * <SkillsLanguage />
 */
export const SkillsLanguage = () => {
  return (
    <section className="mt-12" aria-labelledby="languages-title">
      <div className="card max-w-2xl mx-auto bg-background-main border-2 border-primary-brand/10 shadow-subtle">
        <header className="flex items-center mb-6">
          <Icon
            name="Globe"
            size={20}
            className="mr-3 text-text-main"
            aria-hidden="true"
          />
          <h2
            id="languages-title"
            className="text-xl font-heading font-semibold text-text-main"
          >
            Idiomas
          </h2>
        </header>
        <div className="grid sm:grid-cols-2 gap-6">
          <div
            className="text-center p-4 bg-background-main rounded-medium"
            role="region"
            aria-label="Español"
          >
            <div className="flex justify-center mb-2" aria-hidden="true">
              {[...Array(5)].map((_, index) => (
                <Icon
                  key={index}
                  name="Star"
                  size={14}
                  className="text-success mr-1"
                  aria-hidden="true"
                />
              ))}
            </div>
            <h3 className="font-body font-bold text-text-main mb-1">Español</h3>
            <p className="font-body text-primary-brand text-sm">Nativo</p>
            <p className="font-body text-primary-brand text-xs mt-1">
              Comunicación profesional y técnica fluida
            </p>
          </div>
          <div
            className="text-center p-4 bg-background-main rounded-medium"
            role="region"
            aria-label="Inglés"
          >
            <div className="flex justify-center mb-2" aria-hidden="true">
              {[...Array(5)].map((_, index) => (
                <Icon
                  key={index}
                  name="Star"
                  size={14}
                  className={`mr-1 ${
                    index < 3 ? "text-primary-brand" : "text-secondary"
                  }`}
                  aria-hidden="true"
                />
              ))}
            </div>
            <h3 className="font-body font-bold text-text-main mb-1">Inglés</h3>
            <p className="font-body text-primary-brand text-sm">
              B1 - Intermedio
            </p>
            <p className="font-body text-primary-brand text-xs mt-1">
              Comunicación técnica escrita, reuniones de trabajo
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};
