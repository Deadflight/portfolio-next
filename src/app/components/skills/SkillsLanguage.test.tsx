import React from "react";
import { render, screen } from "@testing-library/react";
import { SkillsLanguage } from "./SkillsLanguage";
import { within } from "@testing-library/react";

// Mock the Icon component
jest.mock("../../../shared/components/Icons/Icons", () => ({
  Icon: ({ name, ...props }: { name: string }) => (
    <span data-testid={`icon-${name}`} {...props} />
  ),
}));

describe("SkillsLanguage", () => {
  beforeEach(() => {
    render(<SkillsLanguage />);
  });

  it("renders the section with the correct title", () => {
    expect(
      screen.getByRole("heading", { name: /idiomas/i, level: 2 })
    ).toBeInTheDocument();
  });

  it("renders Español card with 5 stars", () => {
    const spanishRegion = screen.getByRole("region", { name: "Español" });
    expect(spanishRegion).toBeInTheDocument();
    expect(
      within(spanishRegion).getByRole("heading", { name: "Español", level: 3 })
    ).toBeInTheDocument();
    expect(within(spanishRegion).getByText(/nativo/i)).toBeInTheDocument();
    expect(
      within(spanishRegion).getByText(/comunicación profesional/i)
    ).toBeInTheDocument();
    expect(within(spanishRegion).getAllByTestId("icon-Star")).toHaveLength(5);
  });

  it("renders Inglés card with 5 stars, 3 colored", () => {
    const englishRegion = screen.getByRole("region", { name: "Inglés" });
    expect(englishRegion).toBeInTheDocument();
    expect(
      within(englishRegion).getByRole("heading", { name: "Inglés", level: 3 })
    ).toBeInTheDocument();
    expect(
      within(englishRegion).getByText(/b1 - intermedio/i)
    ).toBeInTheDocument();
    expect(
      within(englishRegion).getByText(/comunicación técnica escrita/i)
    ).toBeInTheDocument();
    const stars = within(englishRegion).getAllByTestId("icon-Star");
    expect(stars).toHaveLength(5);
    // Check className for colored stars
    const coloredStars = stars.filter((star) =>
      (star as HTMLElement).className?.includes("text-text-main")
    );
    expect(coloredStars).toHaveLength(3);
  });

  it("renders the Globe icon in the header", () => {
    expect(screen.getByTestId("icon-Globe")).toBeInTheDocument();
  });

  it("has accessible regions for each language", () => {
    const regions = screen.getAllByRole("region");
    const languageRegions = regions.filter(
      (region) =>
        region.getAttribute("aria-label") === "Español" ||
        region.getAttribute("aria-label") === "Inglés"
    );
    expect(languageRegions).toHaveLength(2);
    expect(languageRegions[0]).toHaveAttribute("aria-label", "Español");
    expect(languageRegions[1]).toHaveAttribute("aria-label", "Inglés");
  });
});
