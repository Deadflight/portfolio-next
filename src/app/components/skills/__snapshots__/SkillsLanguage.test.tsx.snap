// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SkillsLanguage matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="languages-title"
    class="mt-12"
  >
    <div
      class="card max-w-2xl mx-auto bg-background-main border-2 border-primary-brand/10 shadow-subtle"
    >
      <header
        class="flex items-center mb-6"
      >
        <span
          aria-hidden="true"
          class="mr-3 text-text-main"
          data-testid="icon-Globe"
          size="20"
        />
        <h2
          class="text-xl font-heading font-semibold text-text-main"
          id="languages-title"
        >
          Idiomas
        </h2>
      </header>
      <div
        class="grid sm:grid-cols-2 gap-6"
      >
        <div
          aria-label="Español"
          class="text-center p-4 bg-background-main rounded-medium"
          role="region"
        >
          <div
            aria-hidden="true"
            class="flex justify-center mb-2"
          >
            <span
              aria-hidden="true"
              class="text-success-color mr-1"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="text-success-color mr-1"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="text-success-color mr-1"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="text-success-color mr-1"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="text-success-color mr-1"
              data-testid="icon-Star"
              size="14"
            />
          </div>
          <h3
            class="font-body font-bold text-text-main mb-1"
          >
            Español
          </h3>
          <p
            class="font-body text-primary-brand text-sm"
          >
            Nativo
          </p>
          <p
            class="font-body text-primary-brand text-xs mt-1"
          >
            Comunicación profesional y técnica fluida
          </p>
        </div>
        <div
          aria-label="Inglés"
          class="text-center p-4 bg-background-main rounded-medium"
          role="region"
        >
          <div
            aria-hidden="true"
            class="flex justify-center mb-2"
          >
            <span
              aria-hidden="true"
              class="mr-1 text-primary-brand"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="mr-1 text-primary-brand"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="mr-1 text-primary-brand"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="mr-1 text-secondary"
              data-testid="icon-Star"
              size="14"
            />
            <span
              aria-hidden="true"
              class="mr-1 text-secondary"
              data-testid="icon-Star"
              size="14"
            />
          </div>
          <h3
            class="font-body font-bold text-text-main mb-1"
          >
            Inglés
          </h3>
          <p
            class="font-body text-primary-brand text-sm"
          >
            B1 - Intermedio
          </p>
          <p
            class="font-body text-primary-brand text-xs mt-1"
          >
            Comunicación técnica escrita, reuniones de trabajo
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
